lib/term.js

```diff
@@ -1,42 +1,55 @@
 const rdf = require('@rdfjs/data-model')
 
-function term (value, type, datatype) {
-  if (datatype) {
-    datatype = rdf.namedNode(datatype.value || datatype.toString())
+function blankNode (value) {
+  if (value && typeof value !== 'string') {
+    throw new Error('unknown type')
   }
 
-  if (type) {
-    if (type === 'BlankNode') {
-      return rdf.blankNode(value && value.toString())
-    }
+  return rdf.blankNode(value && value.toString())
+}
+
+function literal (value, languageOrDatatype) {
+  if (typeof value !== 'string' && typeof value !== 'number') {
+    throw new Error('unknown type')
```

Comment:
Ditto on vague error.
```suggestion
    throw new Error('The value cannot be converted to a literal node')
```